src/core/qgsvectorfilewriter.cpp

```diff
@@ -982,6 +982,48 @@ QMap<QString, QgsVectorFileWriter::MetaData> QgsVectorFileWriter::initMetaData()
                          )
                        );
 
```

Comment:
Sure, the base could be something like that, with refinements to split the XML option blobs and extract information that is worth to translate from it:
```
from osgeo import ogr
for i in range(ogr.GetDriverCount()):
  drv = ogr.GetDriver(i)
  print(drv.GetMetadata())
```